gammapy/modeling/models/temporal.py

```diff
@@ -679,22 +679,23 @@ def to_table(self):
     def write(self, filename, format="table", overwrite=False):
         """Write a model to disk as per the specified format.
 
-        Parameters:
-            filename : str
-                Name of output file.
-            format : {"table" or "map"}
-                If format is "table", it is serialised as a `~astropy.table.Table`.
-                If "map", then it is serialised as a `~gammapy.maps.RegionNDMap`.
-                Default is "table".
-            overwrite : bool, optional
-                Overwrite existing file. Default is False.
+        Parameters
+        ----------
+        filename : str
+            Name of output file.
+        format : {"table" or "map"}
+            If format is "table", it is serialised as a `~astropy.table.Table`.
+            If "map", then it is serialised as a `~gammapy.maps.RegionNDMap`.
+            Default is "table".
+        overwrite : bool, optional
+            Overwrite existing file. Default is False.
         """
         if self.filename is None:
             raise IOError("Missing filename")
 
         if format == "table":
             table = self.to_table()
-            table.write(filename, overwrite=overwrite)
+            table.write(self.filename, overwrite=overwrite)
```

Comment:
```suggestion
            table.write(filename, overwrite=overwrite)
```
should stay like this but you could add 

```
if filename is None:
    filename=self.filename
```

Reply:
indeed and at the beginning of the function. And this should be in line with l692